### ISSUE TITLE
Implement TCRS subclass of CoordinateReferenceSystem

### ISSUE THREAD

**Issue by prushforth** [Maps4HTML/geoserver#36]
Bind together GeoTools' CoordinateReferenceSystem with GeoWebCache's Gridset, using TCRS parameters, to create a (perhaps) org.geotools.referencing.crs.TiledCoordinateReferenceSystem implementation. 

The objective of that class would be to obtain the benefits of formal CRS handling with the simplicity of creating actual tile caches via the GWC API.  Also to learn more about both.

**Comment by prushforth** [comment]
There's some debate whether a TCRS is a TileMatrixSet or a CRS.

The decision was made early on that TCRS had more in common with CRS than WMTS Tile Matrix Sets, and while one of the authors of that specification was kind enough to create a whole new specification, OGC TMS v1.0.0, in order to provide stable identifiers for TileMatrixSets, that specification has diverged from the intent of TCRS.  Even in version 1.0.0, it specified WebMercatorQuad to be based on the EPSG:4326 CRS, with its latitude,longitude axis order.  This put it at odds with the OSMTILE definition, which relies on the CRS:84 CRS with its longitude,latitude axis order requirement.  

However, TMS v2 has diverged from usefulness even further, because it allows different CRS codes for any given TileMatrixSet, even within an overall "well known" TMS e.g. WebMercatorQuad, which must be determined at run time through [metadata obtained from an HTTP API](https://docs.ogc.org/is/17-083r4/17-083r4.html#toc3). This makes processing instances of TileMatrixSet too unpredictable.

In consequence, TCRS instances should be an instance of a [Compound CRS](https://docs.geotools.org/latest/javadocs/org/opengis/referencing/crs/CompoundCRS.html). In processing coordinates of spatial information, it is vital, essential, paramount to know unequivocally the semantics of the axis values used is.  To know this, programs rely on the axes values being passed to functions as named parameters.  To know which axis value to put into which parameter, the programs must know how to parse the axis values from input coordinates commonly represented as strings.

Axis order of CRS are determined by the CRS definition **by standards decree** aka widespread inviolable cultural convention.  A CRS definition is generally/often communicated by reference, via passing of well known CRS identifiers e.g. "EPSG:4326", that are used to look up local functions for coordinate transformations when appropriate.  The function definitions often have two (or more) axis values as arguments.  The meaning of the arguments is determined by the function (parameter) definitions: sometimes the programmer will have specified that latitude is the first argument and  longitude is the second argument.  When parsing the axis values from inputs, it's vital for the program that invokes the above function definition to KNOW which value it is passing to the function, so that it puts the correct axis in the appropriate argument position.  It KNOWS this fact from the definition of the EPSG:4326 CRS, which (in the particular case of EPSG:4326) specifies that latitude shall come before longitude in parsing otherwise un-marked up coordinate strings, bytes or however that data is encoded. 

In other words, the axis order of EPSG:4326, or any other well-known coordinate system definition **never changes**.  So it should be too, with the axis order of coordinates in MapML TCRS, which is defined by the specification.
